src/pages/home/report/PureReportActionItem.tsx

```diff
@@ -0,0 +1,1113 @@
+
+    // IOUDetails only exists when we are sending money
+    const isSendingMoney =
+        ReportActionsUtils.isMoneyRequestAction(action) &&
+        ReportActionsUtils.getOriginalMessage(action)?.type === CONST.IOU.REPORT_ACTION_TYPE.PAY &&
+        ReportActionsUtils.getOriginalMessage(action)?.IOUDetails;
+
+    const updateHiddenState = useCallback(
+        (isHiddenValue: boolean) => {
+            setIsHidden(isHiddenValue);
+            const message = Array.isArray(action.message) ? action.message?.at(-1) : action.message;
+            const isAttachment = ReportUtils.isReportMessageAttachment(message);
+            if (!isAttachment) {
+                return;
+            }
+            updateHiddenAttachments(action.reportActionID, isHiddenValue);
+        },
+        [action.reportActionID, action.message, updateHiddenAttachments],
+    );
+
+    useEffect(
+        () => () => {
+            // ReportActionContextMenu, EmojiPicker and PopoverReactionList are global components,
+            // we should also hide them when the current component is destroyed
+            if (ReportActionContextMenu.isActiveReportAction(action.reportActionID)) {
+                ReportActionContextMenu.hideContextMenu();
+                ReportActionContextMenu.hideDeleteModal();
+            }
+            if (EmojiPickerAction.isActive(action.reportActionID)) {
+                EmojiPickerAction.hideEmojiPicker(true);
+            }
+            if (reactionListRef?.current?.isActiveReportAction(action.reportActionID)) {
+                reactionListRef?.current?.hideReactionList();
+            }
+        },
+        [action.reportActionID, reactionListRef],
+    );
+
+    useEffect(() => {
+        // We need to hide EmojiPicker when this is a deleted parent action
+        if (!isDeletedParentAction || !EmojiPickerAction.isActive(action.reportActionID)) {
+            return;
+        }
+
+        EmojiPickerAction.hideEmojiPicker(true);
+    }, [isDeletedParentAction, action.reportActionID]);
+
+    useEffect(() => {
+        if (prevDraftMessage !== undefined || draftMessage === undefined) {
+            return;
+        }
+
+        focusComposerWithDelay(textInputRef.current)(true);
+    }, [prevDraftMessage, draftMessage]);
+
+    useEffect(() => {
+        if (!Permissions.canUseLinkPreviews()) {
+            return;
+        }
+
+        const urls = ReportActionsUtils.extractLinksFromMessageHtml(action);
+        if (lodashIsEqual(downloadedPreviews.current, urls) || action.pendingAction === CONST.RED_BRICK_ROAD_PENDING_ACTION.DELETE) {
+            return;
+        }
+
+        downloadedPreviews.current = urls;
+        Report.expandURLPreview(reportID, action.reportActionID);
+    }, [action, reportID]);
+
+    useEffect(() => {
+        if (draftMessage === undefined || !ReportActionsUtils.isDeletedAction(action)) {
+            return;
+        }
+        Report.deleteReportActionDraft(reportID, action);
```

Comment:
Same here. `Report.deleteReportActionDraft` is directly connected to Onyx

Reply:
done